helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -799,7 +799,12 @@ public static boolean verify(Verifier verifier, long timeout) throws Exception {
     long start = System.currentTimeMillis();
     do {
       boolean result = verifier.verify();
-      if (result || (System.currentTimeMillis() - start) > timeout) {
+      boolean isTimedout = (System.currentTimeMillis() - start) > timeout;
+      if (result || isTimedout) {
+        if (isTimedout && !result) {
+          LOG.error("verifier time out, consider try longer timeout, stack trace{}",
+              Arrays.asList(Thread.currentThread().getStackTrace()));
```

Comment:
This stack trace would be printed in one line, right? It'd be difficult to read. How about using 

 new Throwable().printStackTrace();

Reply:
This belongs to https://github.com/apache/helix/pull/1447/files.
The using one line is intentional as we don't want to mix this one with an exception thrown which has multiple lines.